internal/pkg/agent/application/upgrade/step_download.go

```diff
@@ -47,7 +48,12 @@ func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI stri
 		"source_uri", settings.SourceURI, "drop_path", settings.DropPath,
 		"target_path", settings.TargetDirectory, "install_path", settings.InstallPath)
 
-	fetcher, err := newDownloader(version, u.log, &settings)
+	parsedVersion, err := agtversion.ParseVersion(version)
+	if err != nil {
+		return "", fmt.Errorf("error parsing version %q: %w", version, err)
+	}
+
+	fetcher, err := newDownloader(parsedVersion, u.log, &settings)
 	if err != nil {
 		return "", errors.New(err, "initiating fetcher")
```

Comment:
While you are here, I would just go ahead and change from `errors.New(err...)` to `fmt.Errorf`. I mean that through the whole file.

Reply:
Does that mean that the usage of package `github.com/elastic/elastic-agent/internal/pkg/agent/errors` is deprecated ? (I am more than happy if it is and we use the go stdlib error wrapping :tada: )

Reply:
Oh yes, I want it gone!